pkg/provider/docker/docker.go

```diff
@@ -127,34 +137,102 @@ func (p *DockerClassicProvider) GetState(ctx context.Context, name string) (inst
 }
 
 func (p *DockerClassicProvider) NotifyInstanceStopped(ctx context.Context, instance chan<- string) {
-	msgs, errs := p.Client.Events(ctx, types.EventsOptions{
-		Filters: filters.NewArgs(
-			filters.Arg("scope", "local"),
-			filters.Arg("type", string(events.ContainerEventType)),
-			filters.Arg("event", "die"),
-		),
-	})
-	for {
+	reconnectDelay := p.reconnectInitialDelay
```

Comment:
These changes are rather big, and before reviewing this bit, let's try to find some github project that uses the event endpoint and that does reconnect